src/items.cpp

```diff
@@ -326,9 +326,33 @@ bool Items::loadFromXml()
 			parseItemNode(itemNode, id++);
 		}
 	}
+
+	buildInventoryList();
 	return true;
 }
 
+void Items::buildInventoryList()
+{
+	inventory.reserve(30000);
+	for(const auto &type: items) {
+		if(type.weaponType != WEAPON_NONE || type.ammoType != AMMO_NONE ||
+			type.attack != 0 || type.defense != 0 ||
+			type.extraDefense != 0 || type.armor != 0 ||
+			(type.slotPosition & SLOTP_NECKLACE) == SLOTP_NECKLACE ||
+			(type.slotPosition & SLOTP_RING) == SLOTP_RING ||
+			(type.slotPosition & SLOTP_AMMO) == SLOTP_AMMO ||
+			(type.slotPosition & SLOTP_FEET) == SLOTP_FEET ||
+			(type.slotPosition & SLOTP_HEAD) == SLOTP_HEAD ||
+			(type.slotPosition & SLOTP_ARMOR) == SLOTP_ARMOR ||
+			(type.slotPosition & SLOTP_LEGS) == SLOTP_LEGS)
+		{
+			inventory.push_back(type.clientId);
+		}
+	}
+	inventory.shrink_to_fit();
+	std::sort(inventory.begin(), inventory.end());
+}
```

Comment:
It seems I can't use copy_if
```	std::vector<ItemType> type_inventory;
	std::copy_if(items.begin(), items.end(), std::back_inserter(type_inventory), predicateInsertItem);```
A snippet like this will give me some errors because of this `ItemType(const ItemType& other) = delete;`
:(